invisible_cities/reco/icaro_components.py

```diff
@@ -0,0 +1,173 @@
+import numpy        as np
+import pandas       as pd
+
+from   typing              import Optional
+from ..types.symbols       import KrFitFunction
+from .. evm.ic_containers  import FitFunction
+from .. core.fit_functions import polynom, expo
+
+
+def lin_seed(x : np.array,
+             y : np.array):
+
+    '''
+    Estimate the seed for a linear fit.
+
+    Parameters
+    ----------
+    x : np.array
+        Independent variable.
+    y : np.array
+        Dependent variable.
+
+    Returns
+    -------
+    seed : tuple
+        Seed parameters (intercept, slope) for the linear fit.
+    '''
+
+
+    x0, x1 = x.min(), x.max()
+    y0, y1 = y.min(), y.max()
+
+    b = (y1 - y0) / (x1 - x0)
+    a = y0 - b * x0
+
+    seed = a, b
+
+    return seed
+
+
+def expo_seed(x   : np.array,
+              y   : np.array,
+              eps : Optional[float] = 1e-12):
+
+    '''
+    Estimate the seed for an exponential fit.
+
+    Parameters
+    ----------
+    x : np.array
+        Independent variable.
+    y : np.array
+        Dependent variable.
+    eps : float, optional
+        Small value added to prevent division by zero, default is 1e-12.
+
+    Returns
+    -------
+    seed : tuple
+        Seed parameters (constant, mean) for the exponential fit.
+    '''
+
+    x, y  = zip(*sorted(zip(x, y)))
+    const = y[0]
+
+    if y[-1] / (y[0] + eps) <= 0: slope = 0
+    else: slope = (x[-1] - x[0]) / np.log(y[-1] / (y[0] + eps))
```

Comment:
Actually, I think it is better to avoid the `eps` altogether by checking that `y[-1]` and `y[0]` are different. this requires two checks, but it's more robust. There is always a non-zero chance that `y[-1] == y[0] + eps`...